### DISCUSSION TITLE
Downloading MET config files from METplus cloud

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2146] [Training] [answered]
Is it possible to download the Grid-Stat config file for the tutorial from the METplus student EC2 instance for editing? I'm finding it difficult to use the vi editor in that environment, particularly when I try to paste text into the vi session from a web document such as the tutorial content pages. Perhaps there is a better command line editor which will allow easier pasting of text. Thanks.

**Comment by j-opatz** [comment]
Hello,

And thank you for your question! Let me pull in @fisherhucar and see if he can provide some guidance for you.

**Comment by hankenstein2** [comment]
I have a couple of questions, which config files are you looking for? The ones from the online tutorial can be found in the METplus github repository or with the files you can download from the online tutorial.
How are you accessing the EC2 instance? If you are using ssh from a linux/unix or macOS machine you can use scp to bring the config file you want back to your own machine

scp ec2-user@<ip_address>:/d1/METplus-5.0.0-Tutorial/<file_name> .

**Reply by jwraby** [reply]
Thanks for your reply, Hank. I am accessing EC2 from my personal windows machine using PowerShell. I am not familiar with PowerShell, but it was the only way I am allowed right now to connect to the cloud. I normally use ssh to connect to linux/unix machines from my govt laptop and am more comfortable with that. On the govt laptop I use PuTTY to open terminal sessions using ssh, but on my personal laptop all I have is PowerShell. 

The config file I'm working with is the one which was copied per the instructions in the tutorial:
https://dtcenter.org/metplus-practical-session-guide-version-5-0/session-1-metplus-setupgrid-grid/met-tool-grid-stat/configure

I copied the config file per the example given, but when using vi in EC2 (via PowerShell) I can't get the text from the tutorial page to paste into the vi session. This is very odd behavior and not expected based on my experience on linux/unix machines. 

I am familiar with the scp command and will give that a try to copy the config file to my personal laptop and edit it with Notepad and then copy it back to EC2.

**Reply by jwraby** [reply]
I tried using the scp command (sanitized version below)
scp ec2-user@ip-xxx-xx-xx-xxx:/d1/METplus-5.0.0_Tutorial/output/met_output/grid_stat/GridStatConfig_tutorial .
The response is "ssh: Could not resolve hostname ip-xxx-xx-xx-xxx: No such host is known." Not sure what can be done to fix this. Any ideas?

For my next step, I will work on getting the vi within the EC2 instance to accept pasted text like I'm used to in a linux/unix environment. There must be something I'm not doing correctly.

**Comment by hankenstein2** [comment]
Try the solution for cutting and pasting found here and let me know if that works.
https://stackoverflow.com/questions/28679743/copying-and-pasting-text-from-windows-to-a-remote-linux-file

for your scp command try removing the "ip-" part of your command, i.e. just do scp ec2-user@xxx.xx.xx.xxx

**Reply by jwraby** [reply]
Thanks for researching the cutting and pasting solution and the scp suggestion. Will give them and try. Found a different command line editor which, so far,  is allowing intuitive cutting and pasting. Looks promising.

**Reply by jwraby** [reply, marked as answer]
I tried using the suggestion from stackoverflow, but it didn't work for me. I tried removing the "ip-" from the scp command but get the same response:
scp ec2-user@xx-xxx-xx-xxx:/d1/METplus-5.0.0_Tutorial/output/met_output/grid_stat/GridStatConfig_tutorial .
ssh: Could not resolve hostname xx-xxx-xx-xxx: No such host is known.

Based on what I've been able to learn about the Emacs editor which is working in the EC2 instance, I think it will work for me. I can just edit the file without having to copy it to my local machine. The copy/paste seems to work  as I need it to. As long as I can copy the edit changes needed to support the tutorial from the tutorial web page, it looks like this will work. Thanks for your help.
